### PR TITLE
Add documentation for interactive migration

### PR DESCRIPTION
**Pull request by DavidLawes**
## What does this change?
Adds documentation for our plan for interactive migration. 

This is only a documentation change, no actual code change.

### CODE REVIEW

File: docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+To view a pressed page there are a couple of options:
+- Get the document from S3 directly (aws-frontend-archive).
+- Intermediate solution: add the interactive path to the frontend config (https://github.com/guardian/frontend/blob/dlawes/serve-pressed-interactives/common/app/services/dotcomrendering/PressedInteractives.scala#L11).
+- Long-term solution: add tag tracking/dcroptout to article.
+
+**Can we opt-out of pressing and render via frontend or DCR?**
+
+Yes! In the short term, if we want to opt-out of serving pressed content then we omit or remove the interactive path in the relevant frontend config. In the long term, if we want to opt-out of serving the pressed content we will remove, or we will not apply, the chosen tracking tag (tracking/dcroptout).
+
+Note that, after some agreed period of time, I believe our intention would be to remove the rendering code from the frontend codebase, after which point articles could only be rendered via DCR.
```
Comment (nicl) [inline docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md:78]: I would firm this bit up: `I believe our intention` -> `The plan is...`.

Comment (nicl) [reply docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md:78]: Also, how can we ensure this happens? A Trello card might get lost. Perhaps a Dotcom calendar event linked to a Trello card describing the work (and including a link to this doc too)?

And thoughts on a reasonable timeframe? Obviously need to discuss with Rob+Visuals, but 3 months might be a good balance between enough time to provide safety, while also realising the benefits of deleting the code soon as well.

File: docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+**What about interactives via AMP?**
+
+Interactives have to opt-in to rendering via AMP using a specific tag. Interactives rendering via AMP are out of scope for this migration.
+
+About rendering interactives for apps, I believe it’s a separate flow. Given we’re not changing the article content then legacy interactives for mobile will be unaffected. We need to be aware that if there is a request to update the content of a legacy interactive this would need to be done via Composer for the mobile version (so the CAPI response is updated), as well as updating the pressed version for web.
```
Comment (nicl) [inline docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md:84]: Worth adding a heading here for Apps? So as to distinguish from AMP.

**Review by nicl — APPROVED** [review]
This is great - very detailed.

It might be worth adding a separate ADR to DCR. Those are typically more concise so we could summarise and link to this longer document there. (I might attempt this today - will ping if so.)

File: docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md (hunk 2 above)
Comment (DavidLawes) [reply docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md:84]: Yes, will do!

File: docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md (hunk 1 above)
Comment (DavidLawes) [reply docs/06-features-and-components/04-The-Interactive-Librarian/02-interactive-migration.md:78]: Great suggestion - I'll create a trello card and a calendar event, will link both to this doc

**Comment by DavidLawes** [comment]
> This is great - very detailed.
> 
> It might be worth adding a separate ADR to DCR. Those are typically more concise so we could summarise and link to this longer document there. (I might attempt this today - will ping if so.)

Sorry, newbie question, what do we mean by "adding a separate ADR to DCR"? I think we're suggesting adding some equivalent documentation into the dotcom-rendering codebase (maybe inside the doc/contracts folder?)

**Pull request merged**